src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/ReadyToRunGenericHelperNode.cs

```diff
@@ -73,6 +73,39 @@ protected sealed override void OnMarked(NodeFactory factory)
             }
         }
 
+        protected override DependencyList ComputeNonRelocationBasedDependencies(NodeFactory factory)
+        {
+            DependencyList dependencies = new DependencyNodeCore<NodeFactory>.DependencyList();
+
+            // When methods use generic lookup nodes, make sure they're also available as template dictionary nodes
+            if (_dictionaryOwner is MethodDesc)
+            {
+                dependencies.Add(factory.NativeLayout.TemplateMethodLayout((MethodDesc)_dictionaryOwner), "Type loader template");
+                dependencies.Add(_lookupSignature.TemplateDictionaryNode(factory), "Type loader template");
```

Comment:
Ok I moved the TemplateDictionaryNodes to be conditional on the relevant type / method templates. I left the TemplateMethodLayout / TemplateTypeLayout in R2RGenericHelperNode.cs for now since it's currently the best logical place for it imo.